runtime/expr.h

```diff
@@ -441,6 +441,22 @@ template <> SLINKY_ALWAYS_INLINE inline index_t make_binary<logical_and>(index_t
 template <> SLINKY_ALWAYS_INLINE inline index_t make_binary<logical_or>(index_t a, index_t b) { return a || b ? 1 : 0; }
 // clang-format on
 
+// clang-format off
+template <typename T> SLINKY_ALWAYS_INLINE SLINKY_UNIQUE bool binary_op_overflows(index_t a, index_t b) { return false; }
```

Comment:
I think `binary_overflows` would be more consistent with other naming (`make_binary`, `add_overflows`, etc.).